app/models/book.rb

```diff
@@ -10,4 +16,19 @@ class Book < ApplicationRecord
     validates :name , presence: true
     validates :release_date , presence: true
     validates :author , presence: true
+    validate :release_date_before_current_date
+
+
+    scope :with_category_count, -> {
```

Comment:
if you add the relation has many categories through book_categories and call the .count function it will do the exact same query